config/api_router.py

```diff
@@ -1,15 +1,35 @@
 from django.conf import settings
+from django.urls import path
 from rest_framework.routers import DefaultRouter, SimpleRouter
 
-# from bahis_management.desk_modules.api.views import UserViewSet
+from bahis_management.desk.api.views import ModuleTypeViewSet, ModuleViewSet, WorkflowViewSet
+from bahis_management.taxonomies.api.views import (
+    AdministrativeRegionCatchmentView,
+    AdministrativeRegionLevelViewSet,
+    AdministrativeRegionViewSet,
+    TaxonomyViewSet,
+)
 
+# API Routers
 if settings.DEBUG:
     router = DefaultRouter()
 else:
     router = SimpleRouter()
 
-# router.register("users", UserViewSet)
-
+router.register("desk/modules", ModuleViewSet)
+router.register("desk/module-types", ModuleTypeViewSet)
+router.register("desk/workflows", WorkflowViewSet)
+router.register("taxonomy/administrative-regions", AdministrativeRegionViewSet)
+router.register("taxonomy/administrative-region-levels", AdministrativeRegionLevelViewSet)
```

Comment:
@ChasNelson1990 It is  bit confusing. here the route is `administrative` but in [taxonomies\urls](https://github.com/road86/bahis-management/blob/cd829cf11af010635f06f7b9d6c768d49f10f98b/bahis_management/taxonomies/urls.py#L27) 
the route is `adminstrative (without i)`. Is it intentional or a typo? is this why the administrative region migration is not working?

Reply:
It's definitely a typo on my part, my bad. The admin region migration does work fine on my laptop though :confused: but yes, please correct this typo anywhere you see it.

Reply:
> It's definitely a typo on my part, my bad. The admin region migration does work fine on my laptop though 😕 but yes, please correct this typo anywhere you see it.

No worries. I thought the route was on the same level. but one is in API, so it is okay. The administrative region level is populating but administrative region is not working.  I am going deep into the code. Now I am looking here ```    bahis3_admin_regions_df.rename(
        columns={
            "name": "title",
            "value": "id",
            "loc_type": "administrative_region_level",
            "parent": "parent_administrative_region",
        },
        inplace=True,
    )```

You have renamed column `parent_administrative_region` but in the database `parent_administrative_region_id`. I am not sure, is it a bug or not.

Reply:
Ohh. Got the bug, 
```
    bahis3_admin_regions_df[
        "parent_administrative_region_id"
    ] = bahis3_admin_regions_df["parent_administrative_region"].replace(
        -1, np.nan
    )
```
`parent_administrative_region_id` will be `parent_administrative_region` without id

Reply:
Whoops, I think that is a bug left in from the "fast" branch that I shared with you. If you use raw SQL you need the _ID but if you use the API you do not need it because Django's ORM handles it

Reply:
Now migration is working. But it is taking almost one hour to update.